runtime/assembly/models/inherentData.ts

```diff
@@ -0,0 +1,80 @@
+import { CompactInt, Bytes, ScaleString, ByteArray } from 'as-scale-codec';
+import { Constants } from '../constants';
+import { DecodedData } from '../codec/decoded-data';
+
+/**
+ * Class representing InherentData transactions in Substrate runtime
+ */
+
+ export class InherentData {
+    /**
+     * A hashtable (Map, in our case) representing the totality of 
+     * inherent extrinsics included in each block. 
+     */
+    public data: Map<ScaleString, ByteArray>;
+
+    constructor(data: Map<ScaleString, ByteArray>){
+        this.data = data;
+    } 
+
+    /**
+     * SCALE Encodes the InherentData into u8[]
+     */
+    toU8a(): u8[] {
+        let result: u8[] = [];
+        let keys = this.data.keys();
+        const len: CompactInt = new CompactInt(keys.length);
+        result = result.concat(len.toU8a());
+        for (let i = 0; i<keys.length; i++){
+            result = result.concat(keys[i].toU8a().slice(1, Constants.INHERENT_IDENTIFIER_LENGTH + 1));
+            assert(this.data.has(keys[i]), "InherentData: Key doesn't exist in data");
```

Comment:
I think that this assertion is unnecessary.
You are getting the keys by doing `this.data.keys()` and after that, you are asserting that there is a value behind that key. If fore some reason there is no key at that value, there will be a bug in the `map.keys()` function, don't you think?

This is my thinking, maybe you have something other in mind